src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -249,6 +249,8 @@ private DeclarationModifiers MakeModifiers(TypeKind typeKind, DiagnosticBag diag
                     allowedModifiers |= DeclarationModifiers.Static | DeclarationModifiers.Sealed | DeclarationModifiers.Abstract | DeclarationModifiers.Unsafe;
                     break;
                 case TypeKind.Struct:
+                    allowedModifiers |= DeclarationModifiers.Ref | DeclarationModifiers.Unsafe;
```

Comment:
Yes, it was allowed before. That is just to allow using pointer types inside a struct/class/interface. 
I do not think it has any other meaning. The type itself does not become "unsafe".

---
In reply to: [116116652](https://github.com/dotnet/roslyn/pull/19411#discussion_r116116652) [](ancestors = 116116652)